test/functional/terminal/buffer_spec.lua

```diff
@@ -22,14 +22,14 @@ describe(':terminal buffer', function()
 
   it('terminal-mode forces various options', function()
     feed([[<C-\><C-N>]])
-    command('setlocal cursorline cursorcolumn scrolloff=4 sidescrolloff=7')
-    eq({ 1, 1, 4, 7 }, eval('[&l:cursorline, &l:cursorcolumn, &l:scrolloff, &l:sidescrolloff]'))
+    command('setlocal cursorline cursorlineopt=both cursorcolumn scrolloff=4 sidescrolloff=7')
+    eq({ 'both', 1, 1, 4, 7 }, eval('[&l:cursorlineopt, &l:cursorline, &l:cursorcolumn, &l:scrolloff, &l:sidescrolloff]'))
     eq('n', eval('mode()'))
 
     -- Enter terminal-mode ("insert" mode in :terminal).
     feed('i')
     eq('t', eval('mode()'))
-    eq({ 0, 0, 0, 0 }, eval('[&l:cursorline, &l:cursorcolumn, &l:scrolloff, &l:sidescrolloff]'))
+    eq({ 'number', 1, 0, 0, 0 }, eval('[&l:cursorlineopt, &l:cursorline, &l:cursorcolumn, &l:scrolloff, &l:sidescrolloff]'))
```

Comment:
Let's add a test case here for `cursorlineopt=line` and make sure `cursorline` is still disabled.